R/MakeDistMat.R

```diff
@@ -5,15 +5,22 @@
 #' @param dist_metric A distance metric between intensities;  Any function that
 #' takes in two vectors of equal length and produces a scalar
 #' @return Distance matrix with rows/columns named appropriately
-MakeDistMat = function(intensity_mat, dist_metric){
-  col_combos = ncol(intensity_mat) %>% combn(2) %>% t()
+MakeDistMat = function(intensity_mat, dist_metric, symmetric=T){
+  if(!symmetric){
+    col_combos = tibble(x1=1:ncol(intensity_mat),x2=1:ncol(intensity_mat)) %>%
+      expand(x1, x2) %>%
+      filter(x1!=x2) %>%
+      as.matrix()
+  }else{
+    col_combos = ncol(intensity_mat) %>% combn(2) %>% t()
+  }
   dist_mat = matrix(NA, nrow = ncol(intensity_mat), ncol = ncol(intensity_mat))
   dists = map_dbl(1:nrow(col_combos), \(i){
     dist_metric(intensity_mat[,col_combos[i, 1]],
                 intensity_mat[,col_combos[i, 2]])
   })
   dist_mat[col_combos] = dists
-  dist_mat[lower.tri(dist_mat)] = t(dist_mat)[lower.tri(dist_mat)]
+  # if(symmetric==T){dist_mat[lower.tri(dist_mat)] = t(dist_mat)[lower.tri(dist_mat)]}
```

Comment:
nit: just write `if(symmetric)`. Also, why is this commented out?

Reply:
I personally prefer `if(symmetric == T && !symmetric == F)` just to be safe ;)

Jokes aside, my sense was that @mariamasotti1 wanted to make it so that symmetric matrices only display the top half, and asymmetric matrices display both halves (please correct me if I'm wrong Maria). This line forces all matrices to be symmetric- I was going to ask Maria about this but forgot.

Reply:
got it - can we remove the commented out line then?

Reply:
Yeah I wasnt sure what we wanted. But I kind of think why store the entire matrix when it can be defined by just the upper or lower triangular?

Reply:
Makes sense to me!